cmake/modules/SetUpLinux.cmake

```
@@ -158,11 +144,8 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL Intel)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -wd597 -wd1098 -wd1292 -wd1478 -wd3373")
   endif()
 
-  # Select flags.
-  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -fp-model precise -g"          CACHE STRING "Flags for release build with debug info" FORCE)
-  set(CMAKE_CXX_FLAGS_RELEASE        "-O2 -fp-model precise"             CACHE STRING "Flags for release build" FORCE)
-  set(CMAKE_CXX_FLAGS_DEBUG          "-O0 -g"                            CACHE STRING "Flags for a debug build" FORCE)
-  set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O2 -fp-model precise -g"          CACHE STRING "Flags for release build with debug info" FORCE)
-  set(CMAKE_C_FLAGS_RELEASE          "-O2 -fp-model precise"             CACHE STRING "Flags for release build" FORCE)
-  set(CMAKE_C_FLAGS_DEBUG            "-O0 -g"                            CACHE STRING "Flags for a debug build" FORCE)
+  # Augment optimisation flags:
+  foreach(CXX_FLAGS CMAKE_CXX_FLAGS_RELWITHDEBINFO CMAKE_CXX_FLAGS_RELEASE CMAKE_C_FLAGS_RELWITHDEBINFO CMAKE_C_FLAGS_RELEASE)
+    set(${${CXX_FLAGS}} "${${CXX_FLAGS}} -fp-model precise")
+  endforeach()
```

```suggestion
    set(CMAKE_CXX_FLAGS_${BUILD_TYPE} "${CMAKE_CXX_FLAGS_${BUILD_TYPE}} -fp-model precise")
    set(CMAKE_C_FLAGS_${BUILD_TYPE} "${CMAKE_C_FLAGS_${BUILD_TYPE}} -fp-model precise")
```
or
```suggestion
    list(APPEND CMAKE_CXX_FLAGS_${BUILD_TYPE} "-fp-model precise")
    list(APPEND CMAKE_C_FLAGS_${BUILD_TYPE} "-fp-model precise")
```
(and if not, `CXX_FLAGS` sounds like a bad iter var name, maybe `FLAG_VAR`?

---

It's possible, but it's setting less than it was setting before. It might still work, though.

---

Sorry, these are not lists (and cannot be made lists). Need to reject this.